source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -224,6 +227,16 @@ void AuthenticatorImpl::verifyKey() {
     // Remove JWT from headers.
     curr_token_->removeJwt(*headers_);
   }
+  if (set_payload_cb_ && provider.payload_in_metadata()) {
+    try {
+      ProtobufWkt::Struct payload_pb;
+      MessageUtil::loadFromJson(jwt_->payload_str_, payload_pb);
+      set_payload_cb_(jwt_->iss_, payload_pb);
```

Comment:
In a second thought, should we make `payload_in_metadata` as a string, so the key will be specified there? It is more consistent with forward_header configuration. And issuer is already in the payload `iss`. It will be easier to RBAC rules that can do `StringMatcher` over issuer rather than use `PathSegment` (exact match only).

Reply:
Hi @lizan I did not understand.  Could you clarify our proposal?   What format should the payload be stored in dynamic_metadata?  

If we just store the json string payload,  how rbac rules can access its individual field, such as iss,  aud.

Reply:
I meant make [`payload_in_meatadata`](https://github.com/envoyproxy/envoy/pull/4707/files#diff-d483c0b33fc9a4a0e6f0b08138f36994R165) config as a string instead of a bool, like `forward_payload_header`. Then use that string as a key.

The benefit of that is the key will be specified by config, then when you configure MetadataMatcher you can do StringMatch against value of `iss`, but if issuer is struct key it couldn't.

Reply:
Done